### PR TITLE
feat(core): update swift sdk dependency version to 0.26.0

### PR DESCRIPTION
**Pull request by atierian**
## Overview

This is the staging branch for the many updates that need to occur to update our dependency on the AWS SDK for Swift to 0.26.0.

## Description of Changes

- Updates type names, initializers, methods, etc from AWS SDK for Swift and ClientRuntime
- Revamps error handling in accordance with https://github.com/awslabs/aws-sdk-swift/pull/1015 and https://github.com/smithy-lang/smithy-swift/pull/559
- Creates a `UserAgentSettingClientEngine` and uses it for all SDK Client Configurations in order to set the `lib` component of the `User-Agent` header.

---

## PRs
🚧  Pending Work
👀  In Review
✅  Merged into `swift-sdk-0260`

- https://github.com/aws-amplify/amplify-swift/pull/3247 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3251 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3254 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3255 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3256 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3257 ✅  
- https://github.com/aws-amplify/amplify-swift/pull/3258 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3260 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3261 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3264 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3267 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3268 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3262 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3274 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3275 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3276 ✅
- https://github.com/aws-amplify/amplify-swift/pull/3278 ✅  
- https://github.com/aws-amplify/amplify-swift/pull/3280 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3281 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3283 ✅ 
- https://github.com/aws-amplify/amplify-swift/pull/3287 ✅ 

---

## Required Changes
🟦  No changes required
🟢  Non-Functional Changes (e.g. config / dependency version / type renaming)
🔶 Functional Changes

| Target | Changes? |
| ---- | ---- |
| Amplify | 🟦  |
| AmplifyTests | 🟦 |
| | |
| AWSPluginsCore | 🔶 |
| AWSPluginsCoreTests | 🔶  | 
| AWSPluginsTestCommon | 🟢 |
| | |
| AWSAPIPlugin | 🔶 |
| AWSAPIPluginTests | 🔶 |
| APIHostApp - Integration Tests  | 🟦 |  
| | |
| AWSCognitoAuthPlugin | 🔶 |
| AWSCognitoAuthPluginUnitTests | 🔶 |
| AuthHostApp - Integration Tests  | 🔶 | 
| AuthHostedUIApp - Integration Tests  | 🟢 |
| | |
| AWSDataStorePlugin | 🟦 |
| AWSDataStoreCategoryPluginTests | 🟦 |  
| DataStoreHostApp - Integration Tests | 🟦 | 
| | |
| AWSS3StoragePlugin | 🔶 |  
| AWSS3StoragePluginTests | 🔶 |  
| StorageHostApp - Integration Tests | 🔶 |  
| | |
| AWSLocationGeoPlugin | 🔶 |  
| AWSLocationGeoPluginTests | 🔶 |  
| GeoHostApp - Integration Tests | 🟦 |  
| | |
| InternalAWSPinpoint | 🔶 |  
| InternalAWSPinpointUnitTests | 🔶 | 
| | |
| AWSPinpointAnalyticsPlugin | 🔶 |  
| AWSPinpointAnalyticsPluginUnitTests | 🔶 |  
| AnalyticsHostApp - Integration Tests  | 🔶 |
| | |
| AWSPinpointPushNotificationsPlugin | 🔶 |  
| AWSPinpointPushNotificationsPluginUnitTests | 🟦 |  
| PushNotificationsHostApp - Integration Tests  | 🟦 |
| | |
| AWSPredictionsPlugin | 🔶 |  
| AWSPredictionsPluginUnitTests | 🔶 |
| CoreMLPredictionsPlugin | 🟦 |  
| CoreMLPredictionsPluginUnitTests | 🟦 | 
| PredictionsHostApp - Integration Tests | 🔶 |
| | |
| AWSCloudWatchLoggingPlugin | 🔶 |  
| AWSCloudWatchLoggingPluginTests | 🟢 |  
| AWSCloudWatchLoggingPluginHostApp - Integration Tests  | 🟦 |

---

## Open Items

- [x] Remove `FrameworkMetadata` type added in https://github.com/aws-amplify/amplify-swift/pull/3247
- resolved in https://github.com/aws-amplify/amplify-swift/pull/3274
- [x] Add `HttpClientEngine`s everywhere needed to inject user-agent header.  Resolves debt introduced in https://github.com/aws-amplify/amplify-swift/pull/3251
- resolved in https://github.com/aws-amplify/amplify-swift/pull/3274
- [x] There's a new error branch in AWSPredictionsService+Polly that now throws an .unknown error. Take another look at it to see if there's a more appropriate error than can be thrown there. Added in https://github.com/aws-amplify/amplify-swift/pull/3257
- resolved in https://github.com/aws-amplify/amplify-swift/pull/3280
- [x] Double user-agent header value introduced in https://github.com/aws-amplify/amplify-swift/pull/3264
- resolved in https://github.com/aws-amplify/amplify-swift/pull/3274
- [x] Update escape hatch documentation for Swift SDK APIs as necessary.

---

## Documentation Updates
- No documentation updates for Amplify Swift APIs required.
- No documentation updates for escape hatch documentation to AWS SDK for Swift required.

### Escape Hatch Documentation Audit
✅  [Authentication > Escape hatch](https://docs.amplify.aws/lib/auth/escapehatch/q/platform/ios/)
escape_hatch_docs_auth
✅  [Analytics > Escape hatch](https://docs.amplify.aws/lib/analytics/escapehatch/q/platform/ios/)
escape_hatch_docs_analytics
✅  [Geo > Escape hatch](https://docs.amplify.aws/lib/geo/escapehatch/q/platform/ios/)
escape_hatch_docs_geo
✅  [Predictions > Escape hatch](https://docs.amplify.aws/lib/predictions/escapehatch/q/platform/ios/)
escape_hatch_docs_predictions
✅  [Storage > Escape hatch](https://docs.amplify.aws/lib/storage/escapehatch/q/platform/ios/)
escape_hatch_docs_storage
✅  [Logging > Escape hatch](https://docs.amplify.aws/lib/logging/escapehatch/q/platform/ios/)
escape_hatch_docs_logging

---

## General Checklist

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
- [x] New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/AuthenticationProviderDeleteUserTests.swift [hunk 1]
```diff
@@ -75,16 +76,17 @@ class AuthenticationProviderDeleteUserTests: BasePluginTest {
     /// - Then:
     ///    - I should get a .service error with .network as underlying error
     ///
-    func testOfflineDeleteUser() async {
+    func testOfflineDeleteUser() async throws {
+        // TODO: How are client side retry errors now modeled?
```
Comment (phantumcode) [inline AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/AuthenticationProviderDeleteUserTests.swift:80]: does this TODO need to be resolve?

File: AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/AuthenticationProviderDeleteUserTests.swift [hunk 2]
```diff
@@ -110,16 +112,17 @@ class AuthenticationProviderDeleteUserTests: BasePluginTest {
     ///    - I should get a .service error with .network as underlying error for the first call
     ///    - I should get a valid response for the second call
     ///
-    func testOfflineDeleteUserAndRetry() async {
+    func testOfflineDeleteUserAndRetry() async throws {
+        // TODO: How are client side retry errors now modeled?
```
Comment (phantumcode) [inline AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/AuthenticationProviderDeleteUserTests.swift:116]: same comment here, does TODO need to be resolve?

File: AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/AuthenticationProviderDeleteUserTests.swift (hunk 1 above)
Comment (atierian) [reply AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/AuthenticationProviderDeleteUserTests.swift:80]: Thanks for catching this. Addressed in:
- https://github.com/aws-amplify/amplify-swift/pull/3287

File: AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/AuthenticationProviderDeleteUserTests.swift (hunk 2 above)
Comment (atierian) [reply AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/AuthenticationProviderDeleteUserTests.swift]: addressed in [a76284a](https://github.com/aws-amplify/amplify-swift/pull/3248/commits/a76284a55880354a9c5a1b1f289fecf62155cb62)

File: AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/AuthenticationProviderDeleteUserTests.swift (hunk 1 above)
Comment (atierian) [reply AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/AuthenticationProviderDeleteUserTests.swift]: merged in [a76284a](https://github.com/aws-amplify/amplify-swift/pull/3248/commits/a76284a55880354a9c5a1b1f289fecf62155cb62)

**Review by phantumcode — APPROVED** [review]

**Pull request merged**